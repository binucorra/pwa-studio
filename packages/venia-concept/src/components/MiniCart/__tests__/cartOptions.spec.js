import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import CartOptions from '../cartOptions';

jest.mock('src/components/LoadingIndicator', () => {
    return {
        __esModule: true,
        default: () => '( Configurable Loading Indicator Component Here )',
        loadingIndicator: '( Loading Indicator Component Here )'
    };
});
test('it renders correctly', () => {
    const props = {
        cartItem: {
            item_id: 99,
            name: 'cartItem name',
            price: 99,
            qty: 99
        },
        configItem: {
            __typename: 'simple'
        },
        endEditItem: jest.fn(),
        updateCart: jest.fn()
    };
    
    const tree = createTestInstance(
        <CartOptions {...props} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
