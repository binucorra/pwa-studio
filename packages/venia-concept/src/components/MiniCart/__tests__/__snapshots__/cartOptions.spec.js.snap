// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div>
    <span>
      cartItem name
    </span>
    <span>
      <div />
    </span>
  </div>
  <div>
    <section>
      <h2>
        <span>
          Quantity
        </span>
      </h2>
      <div>
        <span
          style={
            Object {
              "--iconsAfter": 1,
              "--iconsBefore": 0,
            }
          }
        >
          <span>
            <select
              aria-label="product's quantity"
              name="quantity"
              onBlur={[Function]}
              onChange={[Function]}
              value={99}
            >
              <option
                value={1}
              >
                1
              </option>
              <option
                value={2}
              >
                2
              </option>
              <option
                value={3}
              >
                3
              </option>
              <option
                value={4}
              >
                4
              </option>
            </select>
          </span>
          <span />
          <span>
            <span>
              <svg
                fill="none"
                height={18}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width={18}
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="6 9 12 15 18 9"
                />
              </svg>
            </span>
          </span>
        </span>
        <p />
      </div>
    </section>
  </div>
  <div>
    <button
      onClick={[MockFunction]}
      type="button"
    >
      <span>
        <span>
          Cancel
        </span>
      </span>
    </button>
    <button
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span>
        <span>
          Update Cart
        </span>
      </span>
    </button>
  </div>
  <div>
    ( Configurable Loading Indicator Component Here )
  </div>
</form>
`;
